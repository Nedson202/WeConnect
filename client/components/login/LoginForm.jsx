import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import TextField from '../common/TextField';
import passwordToggler from '../../utils/passwordToggler';
import '../../index.scss';

/**
 * @class LoginForm
 *
 * @extends {Component}
 */
const LoginForm = ({
  state, onChange, onSubmit, isLoading
}) => {
  const { username, password } = state;
  return (
    <div>
      <form onSubmit={onSubmit}>
        <div className="custom-form-style">
          <h3 className="text-center form-header">Login</h3>
          <TextField
            label="Username"
            onChange={onChange}
            value={username}
            field="username"
            placeholder="Username"
            type="text"
          />
          <div className="form-group col-md-6 offset-md-3 col-lg-8 offset-lg-2">
            <label
              id="control-label"
            >Password
            </label>
            <input
              value={password}
              onChange={onChange}
              type="password"
              name="password"
              id="password"
              className="form-control"
              placeholder="password"
            />
            <div className="input-group-btn">
              <li className="btn btn-default" onClick={passwordToggler()}>
                <i className="fa fa-eye" aria-hidden="true" id="add-hide" />
                <i className="fa fa-eye-slash hide" aria-hidden="true" id="remove-hide" />
              </li>
            </div>
          </div>
          <button
            disabled={isLoading}
            className="btn btn-outline-success"
            type="submit"
            id="submit-button"
          >
            { isLoading ? (
              <span className="processing-info">
                  processing <i className="fa fa-spinner fa-spin" />
              </span>) : <span><i className="fa fa-paper-plane" />&nbsp; Login</span>}
          </button>
          <p className="text-center account-block">Don&apos;t have an account? &nbsp;
            <Link to="/signup" className="link">signup</Link>
          </p>
        </div>
      </form>
    </div>
  );
};

LoginForm.propTypes = {
  onChange: PropTypes.func.isRequired,
  onSubmit: PropTypes.func.isRequired,
  state: PropTypes.object.isRequired,
  isLoading: PropTypes.bool.isRequired,
};

export default LoginForm;
