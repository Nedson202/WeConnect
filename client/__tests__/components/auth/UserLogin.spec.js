import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import thunk from 'redux-thunk';
import configureMockStore from 'redux-mock-store';
import Adapter from 'enzyme-adapter-react-16';
import toJSON from 'enzyme-to-json';
import sinon from 'sinon';
import ConnectedLogin, { Login } from '../../../components/login/Login';
import LoginForm from '../../../components/login/LoginForm';
import state from '../../__mockData__/userData';

const middlewares = [thunk];
const mockStore = configureMockStore(middlewares);
const { loginState } = state;

configure({ adapter: new Adapter() });
localStorage = {};

const props = {
  userLoginRequest: jest.fn(() => Promise.resolve()),
  history: {},
  isLoading: false,
  onChange: jest.fn(),
  onSubmit: jest.fn(),
  state: loginState
};

const setup = () => {
  return shallow(<Login {...props} />);
};

const event = (name, value) => ({
  preventDefault: jest.fn(),
  target: {
    name,
    value
  }
});

const userData = {
  username: 'test',
  password: 'test@test.com'
}

describe('User login component', () => {
  beforeEach(() => {
    global.document = {
      title: () => {}
    }
  })

  it('should render login component', () => {
    const wrapper = setup();
    expect(wrapper.find('div').length).toBe(2);
    expect(wrapper.find('LoginForm').length).toBe(1);
    expect(toJSON(wrapper)).toMatchSnapshot();
  }) 

  it('should render login form', () => {
    const wrapper = shallow(<LoginForm {...props} />);
    expect(wrapper.find('div').length).toBe(4);
    expect(wrapper.find('form').length).toBe(1);
    expect(wrapper.find('Link').length).toBe(1);
    expect(wrapper.find('button').length).toBe(1);
    expect(wrapper.find('i').length).toBe(3);
    expect(wrapper.find('span').length).toBe(1);
    expect(wrapper.find('button').simulate('click'));
    wrapper.setProps({ isLoading: true });
    expect(wrapper.find('span').length).toBe(1);
    expect(wrapper.find('TextField').length).toBe(1);
    expect(toJSON(wrapper)).toMatchSnapshot();
  }) 

  it('should render connected login', () => {
    const store = mockStore({});
    const wrapper = shallow(<connectedLogin store={store} />);
    expect(wrapper.length).toBe(1);
  }) 
});

describe('Login onchange method', () => {
  beforeEach(() => {
    global.document = {
      title: () => {}
    }
  });
  
  
  it('should simulate input change', () => {
    const wrapper = setup();
    const loginForm = wrapper.find('LoginForm');
    loginForm.simulate('change', event('username', userData.username));
    expect(wrapper.instance().state.username).toBe(userData.username);
    loginForm.simulate('change', event('password', userData.password));
    expect(wrapper.instance().state.password).toBe(userData.password);
  });
});

describe('Submit function', () => {
  it('should login user when user details is set to the state', () => {
    const wrapper = setup();
    const loginForm = wrapper.find('LoginForm');
    // const button = loginForm.find('#submit-button');
    wrapper.setState(userData);

    loginForm.simulate('submit', event());
  });
});

