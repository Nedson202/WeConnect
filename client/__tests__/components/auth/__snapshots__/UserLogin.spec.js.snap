// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User login component should render login component 1`] = `
<div>
  <div
    id="background-image"
  />
  <LoginForm
    isLoading={false}
    onChange={[Function]}
    onSubmit={[Function]}
    state={
      Object {
        "errors": Object {},
        "password": "",
        "username": "",
      }
    }
    userLoginRequest={[MockFunction]}
  />
</div>
`;

exports[`User login component should render login form 1`] = `
<div>
  <form
    onSubmit={[MockFunction]}
  >
    <div
      className="custom-form-style"
    >
      <h3
        className="text-center form-header"
      >
        Login
      </h3>
      <TextField
        field="username"
        label="Username"
        onChange={[MockFunction]}
        placeholder="Username"
        type="text"
        value="mike"
      />
      <div
        className="form-group col-md-6 offset-md-3 col-lg-8 offset-lg-2"
      >
        <label
          id="control-label"
        >
          Password
        </label>
        <input
          className="form-control"
          id="password"
          name="password"
          onChange={[MockFunction]}
          placeholder="password"
          type="password"
          value="alen@stuart.mill"
        />
        <div
          className="input-group-btn"
        >
          <li
            className="btn btn-default"
            onClick={[Function]}
          >
            <i
              aria-hidden="true"
              className="fa fa-eye"
              id="add-hide"
            />
            <i
              aria-hidden="true"
              className="fa fa-eye-slash hide"
              id="remove-hide"
            />
          </li>
        </div>
      </div>
      <button
        className="btn btn-outline-success"
        disabled={true}
        id="submit-button"
        type="submit"
      >
        <span
          className="processing-info"
        >
          processing 
          <i
            className="fa fa-spinner fa-spin"
          />
        </span>
      </button>
      <p
        className="text-center account-block"
      >
        Don't have an account? Â 
        <Link
          className="link"
          replace={false}
          to="/signup"
        >
          signup
        </Link>
      </p>
    </div>
  </form>
</div>
`;
